src/shared/sam/debugger/javaSamDebug.ts

```diff
@@ -16,3 +21,42 @@ export async function activateJavaExtensionIfInstalled() {
         await extension.activate()
     }
 }
+
+export async function makeJavaConfig(config: SamLaunchRequestArgs): Promise<SamLaunchRequestArgs> {
+    if (!config.baseBuildDir) {
+        throw Error('invalid state: config.baseBuildDir was not set')
+    }
+
+    config = {
+        ...config,
+        request: config.noDebug ? 'launch' : 'attach',
+    }
+
+    config.codeRoot = getCodeRoot(config.workspaceFolder, config)!
+    config.templatePath = await makeInputTemplate(config)
+
+    config.type = 'java'
+    config.runtimeFamily = RuntimeFamily.Java
+
+    if (!config.noDebug) {
+        config.hostName = '127.0.0.1'
+        config.port = config.debugPort
+        if (isImageLambdaConfig(config)) {
+            config.containerEnvVars = {
+                _JAVA_OPTIONS:
+                    config.runtime === 'java11'
+                        ? `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=*:${config.debugPort} -XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC -XX:-TieredCompilation -Djava.net.preferIPv4Stack=true`
+                        : `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=${config.debugPort} -XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC -XX:-TieredCompilation -Djava.net.preferIPv4Stack=true -Xshare:off`,
+            }
+        }
+    }
+
+    return config
+}
+
+export async function invokeJavaLambda(ctx: ExtContext, config: SamLaunchRequestArgs): Promise<SamLaunchRequestArgs> {
+    config.samLocalInvokeCommand = new DefaultSamLocalInvokeCommand([WAIT_FOR_DEBUGGER_MESSAGES.JAVA])
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    config.onWillAttachDebugger = waitForPort
```

Comment:
if `Picked up _JAVA_OPTIONS` is not a reliable signal,  maybe this should use the same wait function as python
```suggestion
    config.onWillAttachDebugger = waitForPythonDebugAdapter
```